core/src/app.rs

```diff
@@ -614,7 +781,67 @@ impl App {
                         (mut e, Some(v), _) => {
                             // Finally update the environment with the block execution result and
                             // prepare the next loop `envs` object.
-                            e.state.insert(name.clone(), v);
+                            match current_while {
+                                // We're outside a `while` loop, simply set the value.
+                                None => {
+                                    // Note that if the block type is `reduce` this will override in
+                                    // the state the value for the `map` type since they have the
+                                    // same name. This is fine as this is not super useful
```

Comment:
🤔  I'm not sure I am following

Is this expected for example ?
<img width="991" alt="Screenshot 2023-04-13 at 12 02 38" src="https://user-images.githubusercontent.com/14199823/231726176-e1202493-a7d4-4c6b-88e4-044486036c28.png">

Reply:
Argh you're right we don't enforce it. It overwrites. And this is expected. But we don't want that for while/end and map/reduce so that we preserve the output of map and while (which are to some extent useful) vs overriding them with the outputs of reduce/end which are empty.

Does that make sense?